### ISSUE TITLE
Integrating the claim journey with the TRS API client

### ISSUE THREAD

**Issue by peteryates** [DFE-Digital/ecf2#288]
The claim journey was added in #265 with some `FIXME` and `TODO` notes for API client integration.  The API client was just added in in #250 so we can now integrate them.

### Tasks

* [ ] replace the contents of [find_matching_record_in_trs](https://github.com/DFE-Digital/ecf2/blob/main/app/services/appropriate_bodies/claim_an_ect/find_ect.rb#L33) with an actual API call.
* [ ] work out which fields from the example response below we want to show on the 'Check ECT' page. Where possible store them as fields on `PendingInductionSubmission`. We need to work out what data we want to show for the array fields: 
    * [ ] initialTeacherTraining
    * [ ] alerts - thoughts welcome @emily-prudence-dfe 
* [ ] display the extra data on the 'check ECT' page. The final designs haven't yet been discussed so just showing it as a summary list should be fine for now.

```json
{
  "alerts": [
    {
      "alertId": "3fa85f64-5717-4562-b3fc-2c963f66afa6",
      "alertType": {
        "alertTypeId": "3fa85f64-5717-4562-b3fc-2c963f66afa6",
        "alertCategory": {
          "alertCategoryId": "3fa85f64-5717-4562-b3fc-2c963f66afa6",
          "name": "string"
        },
        "name": "string"
      },
      "startDate": "2024-09-18",
      "endDate": "2024-09-18"
    }
  ],
  "induction": {
    "startDate": "2024-09-18",
    "endDate": "2024-09-18",
    "status": "Exempt",
    "statusDescription": "string",
    "certificateUrl": "string"
  },
  "trn": "string",
  "firstName": "string",
  "middleName": "string",
  "lastName": "string",
  "dateOfBirth": "2024-09-18",
  "nationalInsuranceNumber": "string",
  "pendingNameChange": true,
  "pendingDateOfBirthChange": true,
  "emailAddress": "string",
  "qts": {
    "awarded": "2024-09-18",
    "certificateUrl": "string",
    "statusDescription": "string"
  },
  "eyts": {
    "awarded": "2024-09-18",
    "certificateUrl": "string",
    "statusDescription": "string"
  },
  "initialTeacherTraining": [
    {
      "qualification": {
        "name": "string"
      },
      "ageRange": {
        "description": "string"
      },
      "provider": {
        "name": "string",
        "ukprn": "string"
      },
      "subjects": [
        {
          "code": "string",
          "name": "string"
        }
      ],
      "startDate": "2024-09-18",
      "endDate": "2024-09-18",
      "programmeType": "Apprenticeship",
      "programmeTypeDescription": "string",
      "result": "Pass"
    }
  ],
  "npqQualifications": [
    {
      "type": {
        "code": "NPQEL",
        "name": "string"
      },
      "awarded": "2024-09-18",
      "certificateUrl": "string"
    }
  ],
  "mandatoryQualifications": [
    {
      "awarded": "2024-09-18",
      "specialism": "string"
    }
  ],
  "higherEducationQualifications": [
    {
      "subjects": [
        {
          "code": "string",
          "name": "string"
        }
      ],
      "name": "string",
      "awarded": "2024-09-18"
    }
  ],
  "previousNames": [
    {
      "firstName": "string",
      "middleName": "string",
      "lastName": "string"
    }
  ],
  "allowIdSignInWithProhibitions": true
}
```

**Comment by claire-hughez** [comment]
Catch up with Emily P on this ticket today. Just on fields we want to display.

**Comment by emily-prudence-dfe** [comment]
When claiming an ECT, ABs currently see this record, which includes many fields on the DQT including ITT data and QTS status.

<img width="425" alt="image" src="https://github.com/user-attachments/assets/cdaa2895-f095-47a9-920f-ca62172d344a">

We assume that ECTs don't need to see all of this data, but some is important to help ABs fulfil their responsibilities. **For our first iteration, we will show a minimum set of data based on some assumptions on what these responsibilities are - we know we will need to build on this later**

According to policy, the key responsibilities an AB has when an ECT starts induction are:

- Verifying pre-employment checks which the school has undertaken
- Checking that the ECT is eligible to serve an induction (key factors are that they have QTS awarded, that they haven't served an induction prior, and that they are not exempt from serving induction)
- Checking that the organisation can act as the AB for the ECT's induction (organisations which provide AB services may also be ITT providers, and are not eligible to provide AB services for an ECT to whom they also recommended QTS be awarded)

Therefore, we can make some assumptions about the minimum set of data to be displayed here:

- Flagging that an ECT has an alert against them helps ABs to know when to follow up with schools about an ECT's suitability for induction -> **display a flag if an ECT has one or more alerts**
- We will build in system checks for QTS and induction status when we build unhappy paths, so assuming this is the case ABs don't actually need to see this information - but we assume seeing some data on the ECT's page when claiming them is helpful -> **display induction status and start date, QTS status and award date**
- ABs can also claim ECTs who have already served some induction -> **display induction history & extension period**
- ABs need to be able to trace which ITT provider an ECT served their induction with -> **display ITT provider & completion date**

To summarise, within the "check ECT's record" part of the claim service, we should start by showing:

- Induction status and start date (which may be blank)
- Induction history
- _Note: we should make the above consistent with View an ECT's record_
- QTS status and award date
- Alerts flag (Has Alerts yes/no - but no detail on alert type for now)
- ITT provider and ITT completion date

**Comment by gracecbrennan** [comment]
@peteryates to code review.
